include/pistache/base64.h

```diff
@@ -52,7 +45,7 @@ class Base64Decoder
 protected:
     // Convert an octet character to corresponding sextet, provided it can
     //  safely be represented as such. Otherwise return 0xff...
-    std::byte DecodeCharacter(const unsigned char Character) const;
+    std::byte DecodeCharacter(unsigned char Character) const;
```

Comment:
Yes, `const` in function declarations has no effect. It does not harm either, but (in my opinion) it simplifies the function signature, slightly improving readability

Reply:
When you say it has no effect, what do you mean? The effect is the parameter is set as read only throughout its lifetime (the method body), which was what was intended.

Reply:
Sorry, I've been too vague. `const` in function _declarations_ has only effect if the parameter is passed by reference or is a pointer, while const-qualifying a parameter passed by value in function declarations has no effect. `const` on parameters passed by value has only meaning in function _definitions_. 

This also makes sense if you think about it; from the users' perspective, when they look at the function interface (declaration) they don't care whether the function modifies or not a parameter passed by value, as they can pass const and non-const variables because they get copied anyway. On the other hand, they are interested in seeing if references and pointers are const-qualified in the function declarations, because if they are not they cannot pass unmodifiable variables to that function. 

`const` on passed by value parameters is an implementation detail, an should be left in function definitions.

[Here](https://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html)'s the short description of the clang-tidy rule

Reply:
You are right that the `const` on a parameter passed by value in the definition doesn't create an overload from the corresponding declaration that doesn't contain it. But for sake of not having two different signatures throughout the code base for the same method, please keep them consistent.